From 028e13aeffb947f7189b7e06b44cd105f104ef19 Mon Sep 17 00:00:00 2001
From: Geoff Garside <geoff@geoffgarside.co.uk>
Date: Tue, 18 Nov 2008 15:14:55 +0000
Subject: [PATCH] Reorder the way in which map.resource routes are added to the set.
 This prevents the singular named route from hitting :create instead of :show.

---
 actionpack/lib/action_controller/resources.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_controller/resources.rb b/actionpack/lib/action_controller/resources.rb
index b5ea764..c170528 100644
--- a/actionpack/lib/action_controller/resources.rb
+++ b/actionpack/lib/action_controller/resources.rb
@@ -535,9 +535,9 @@ module ActionController
 
         with_options :controller => resource.controller do |map|
           map_collection_actions(map, resource)
-          map_default_singleton_actions(map, resource)
           map_new_actions(map, resource)
           map_member_actions(map, resource)
+          map_default_singleton_actions(map, resource)
 
           map_associations(resource, options)
 
-- 
1.6.0.3


From 163a0e73f54f91492a1bfcf49843e41afeab861b Mon Sep 17 00:00:00 2001
From: Geoff Garside <geoff@geoffgarside.co.uk>
Date: Mon, 24 Nov 2008 12:23:27 +0000
Subject: [PATCH] Test default singleton resource route to ensure it uses GET.
 This is important if using map.root :resource instead of map.root :resources for some reason.

---
 actionpack/test/controller/resources_test.rb |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/actionpack/test/controller/resources_test.rb b/actionpack/test/controller/resources_test.rb
index 541e827..79b28c0 100644
--- a/actionpack/test/controller/resources_test.rb
+++ b/actionpack/test/controller/resources_test.rb
@@ -997,6 +997,16 @@ class ResourcesTest < ActionController::TestCase
     end
   end
 
+  def test_default_singleton_restful_route_uses_get
+    with_routing do |set|
+      set.draw do |map|
+        map.resource :product
+      end
+
+      assert_equal :get, set.named_routes.routes[:product].conditions[:method]
+    end
+  end
+
   protected
     def with_restful_routing(*args)
       with_routing do |set|
-- 
1.6.0.3

